machine_learning/word_frequency_functions.py

```diff
@@ -0,0 +1,127 @@
+    as tf*log10(N/df). It compares the number of times
+    a term appears in a document with the number of documents
+    the term appears in. If df is 0, a ZeroDivisionError will be thrown.
+"""
+
+
+def term_frequency(term : str, document : str) -> int:
+    """
+    A function that returns the number of times a term occurs within
+    a given document.
+    @params: term, the term to search a document for, and document,
+            the document to search within
+    @returns: an integer representing the number of times a term is
+            found within the document
+
+    @examples:
+    >>> term_frequency("to", "To be, or not to be")
+    2
+    """
+    # strip all punctuation and newlines and replace it with ''
+    document_without_punctuation = document.translate(
+        str.maketrans("", "", string.punctuation)
+    ).replace("\n", "")
+    tokenize_document = document_without_punctuation.split(" ")  # word tokenization
+    term_frequency = len(
+        [word for word in tokenize_document if word.lower() == term.lower()]
+    )
+    return term_frequency
+
+
+def document_frequency(term: str, corpus: str) -> int:
+    """
+    A function that calculates the number of documents in a corpus that contain a
+    given term
+    @params : term, the term to search each document for, and corpus, a collection of
+             documents. Each document should be separated by a newline.
+    @returns : the number of documents in the corpus that contain the term you are
+               searching for and the number of documents in the corpus
+    @examples :
+    >>> document_frequency("first", "This is the first document in the corpus.\\nThIs\
+is the second document in the corpus.\\nTHIS is \
+the third document in the corpus.")
+    (1, 3)
+    """
+    corpus_without_punctuation = corpus.translate(
+        str.maketrans("", "", string.punctuation)
+    )  # strip all punctuation and replace it with ''
+    documents = corpus_without_punctuation.split("\n")
+    lowercase_documents = [document.lower() for document in documents]
+    document_frequency = len(
+        [document for document in lowercase_documents if term.lower() in document]
+    )  # number of documents that contain the term
+    return document_frequency, len(documents)
+
+
+def inverse_document_frequency(df : int, N: int) -> float:
+    """
+    A function that returns an integer denoting the importance
+    of a word. This measure of importance is
+    calculated by log10(N/df), where N is the
+    number of documents and df is
+    the Document Frequency.
+    @params : df, the Document Frequency, and N,
+    the number of documents in the corpus.
+    @returns : log10(N/df)
+    @examples :
+    >>> inverse_document_frequency(1, 3)
+    0.477
+    """
+    try:
+        idf = round(log10(N / df), 3)
+        return idf
+    except ZeroDivisionError:
+        print("The term you searched for is not in the corpus.")
+
```

Comment:
if raise this function not returning float

Reply:
what do you mean?

Reply:
If error raised `None` will be returned. Is it good way? Wouldn't be better to check df value before calculation and raise error if `df==0`?. Also please add doctest with this situation.

Reply:
Yes.  Algorithm functions should not print as discussed in CONTRIBUTING.md.  If our type hints promise that we will return a float then we should do so or raise an exception explaining why we can not do so.